### PR TITLE
LPS 181193

### PR DESCRIPTION
**Pull request by rotty3000**
- LPS-181193 [batch] cleanup jsont -> batch-engine-data.json
- LPS-178925 add baseURL property to OAuth2 app configurations
- LPS-181194 [wab] renaming to match documentation: static -> frontend
- LPS-181195 [workspace] set the force policy for OSGi configurations by default (delete on uninstall bundle, update config on bundle update)
- LPS-180491 Make visible configurations which make use of ConfigurationPlugin(s) to inject the companyId property
- LPS-181197 [samples] add missing on-prem compat properties
- LPS-181198 [samples] update batch-engine-data to match exact output of Import/Export Center
- LPS-181199 [batch] make sure bundle updates also cause batch to be executed
- LPS-181200 [workspace] allow properties to be set in client-extension.yaml and add a couple of new utility properties

### CODE REVIEW

**Comment by rotty3000** [comment]
cc @gamerson @drewbrokke

**Comment by rotty3000** [comment]
ci:reopen

File: modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/internal/client/extension/ClientExtension.java [hunk 1]
```diff
@@ -46,13 +46,25 @@ public Map<String, Object> toJSONMap(String pid) {
...
 		Map<String, Object> configMap = new HashMap<>();
 
-		configMap.put("baseURL", "${portalURL}/o/" + projectName);
+		configMap.put(
+			"baseURL",
+			typeSettings.getOrDefault(
+				"baseURL", "${portalURL}/o/" + projectName));
 		configMap.put("description", description);
-		configMap.put("dxp.lxc.liferay.com.virtualInstanceId", "default");
+		configMap.put(
```
Comment (gamerson) [inline modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/internal/client/extension/ClientExtension.java:54]: Hey Ray, I proposed handling this property in a different way here: https://github.com/brianchandotcom/liferay-portal/pull/133078
specifically this way:
https://github.com/brianchandotcom/liferay-portal/pull/133078/files#diff-5e3deeb76842bbeeb314a4735f3a0fecf5e96811d81e7cfa3563ffcb38ad08e2R97

I think we could remove this code where it grabs it from typesettings and instead just let jackson mapping handle it.

Comment (rotty3000) [reply modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/internal/client/extension/ClientExtension.java:54]: Devil's advocate here... do we keep mapping properties (you can see I added several more)?

I'm a little remiss to have 2 different ways of handling all these properties. 

I'll change to that pattern and ideally it'll merge cleanly.

**Pull request closed without merging**